### PR TITLE
Minify js and json

### LINKED ISSUE #41: JSON support
## Specific Demand

It would be nice if manganis minified json automatically.

## Implement Suggestion

I can't think of any useful settings for json. We can just implement automatic minification for the `.json` extension with the `mg!(file("path.json"))` syntax

### PR DESCRIPTION
**Pull request by ealmloff**
This PR builds off of #35 to minify js and json with manganis

Closes #35 
Closes #41

### CODE REVIEW

File: cli-support/src/file.rs [hunk 1]
```diff
@@ -182,3 +195,78 @@ pub(crate) fn minify_css(css: &str) -> String {
...
+                    manganis_common::FileSource::Remote(url) => FileName::Url(url.clone()),
+                });
+                let fm = cm.new_source_file(filename, js.to_string());
+
+                c.minify(
+                    fm,
+                    handler,
+                    &JsMinifyOptions {
+                        compress: BoolOrDataConfig::from_bool(true),
+                        mangle: BoolOrDataConfig::from_bool(true),
```
Comment (DogeDark) [inline cli-support/src/file.rs:218]: Would this mangle function names? If so, I can see situations where we want to preserve them. E.x. calling your function from an `eval`

File: common/src/manifest.rs [hunk 2]
```diff
@@ -42,6 +42,15 @@ impl AssetManifest {
                             }
                         }
                     }
+                    crate::FileOptions::Js(js_options) => {
+                        if js_options.preload() {
+                            if let Ok(asset_path) = file.served_location() {
+                                head.push_str(&format!(
+                                    "<link rel=\"preload\" as=\"script\" href=\"{asset_path}\">\n"
```
Comment (DogeDark) [inline common/src/manifest.rs:49]: We'd probably want a preload setting on the new head elements in the future. Would likely need the linker to propagate.

File: cli-support/src/file.rs (hunk 1 above)
Comment (ealmloff) [reply cli-support/src/file.rs:218]: Anything you can access outside of the module like exported function names or functions attached to the window will not get mangled

File: common/src/manifest.rs (hunk 2 above)
Comment (ealmloff) [reply common/src/manifest.rs:49]: You can use the link head element for preloading assets dynamically in fullstack/ssg:
```rust
head::Link { rel: "preload", as: "script", href: "bla" }
```

**Pull request merged**